app/src/main/java/studyBuddy/Session.java

```diff
@@ -0,0 +1,133 @@
+package studyBuddy;
+
+import java.sql.Timestamp;
+
+public class Session {
+    private Timestamp startTime;
+    private Timestamp endTime;
+    private String name;
+    // In minutes
+    private double expectedTime;
+    private double productiveTime;
+    private double totalTime;
+    // Expected value between 0 and 1
+    private double percentProductive;
+    // TODO: fill this in correctly according to implementation of Time Management module
+    // private TimeManagement timeManager;
+    private boolean sessionOngoing;
+
+    /**
+     * Constructs a new studyBuddy.Session
+     * Parameters are initialized to somewhat meaningless values
+     * Session needs to be started by user
+     */
+    public Session() {
+        startTime = null;
+        endTime = null;
+        name = null;
+        expectedTime = 0.0;
+        productiveTime = 0.0;
+        totalTime = 0.0;
+        // assume they were productive the whole time
+        percentProductive = 1.0;
+        expectedTime = 0.0;
+        // timeManager = null;
+        sessionOngoing = false;
+    }
+
+//    TODO
+//    public void setTimeManager(TimeManagement manager);
+
+    /**
+     * Starts a session, triggered by user interaction
+     * Sets the session name and expected duration as given by user
+     * Records the real time session start time
+     * @param sessionName the name of the task for this session
+     * @param expectedSessionTime the expected duration of this session /
+     *                            expected time to complete task
+     */
+    public void startSession(String sessionName, double expectedSessionTime) {
+        long currentTime = System.currentTimeMillis();
+        startTime = new Timestamp(currentTime);
+        name = sessionName;
+        expectedTime = expectedSessionTime;
+        sessionOngoing = true;
+    }
+
+    /**
+     * Preconditions: session must be ongoing (start session must have been called)
```

Comment:
Given that the sessionOngoing is being tracked we should perform the functionality in endSession if it is set and do nothing otherwise to remove this precondition

Reply:
there's some other cases where this precondition exists and this flag should be used to handle those as well